src/supervise-daemon/supervise-daemon.c

```diff
@@ -1134,7 +1177,13 @@ int main(int argc, char **argv)
 			parse_schedule(applet, NULL, sig);
 
 		str = rc_service_value_get(svcname, "respawn_delay");
-		sscanf(str, "%d", &respawn_delay);
+		if (!str || sscanf(str, "%zd:%zd", &respawn_delay.tv_sec, &respawn_delay.tv_nsec) != 2)
```

Comment:
%zd is for `size_t`. Neither tv_sec or tv_nsec is of that type so this could potentially do OOB write.

For `nsec` you'd need `%ld`. `time_t` doesn't have any printf specifier so you can't directly scan into it, you'd need to use some intermediate variable to scan into and then assign.

Also wouldn't it be better to store the value string directly and reparse with `parse_respawn`? It would make a new version of supervise-daemon be able to read the value written by an older version and be backwards compatible.